main/options.c

```diff
@@ -2363,6 +2365,22 @@ static void processMaxRecursionDepthOption (const char *const option, const char
 	Option.maxRecursionDepth = atol(parameter);
 }
 
+static void processPatternLengthLimit(const char *const option, const char *const parameter)
+{
+	char* end;
+	long patternLengthLimit;
+
+	if (parameter == NULL || parameter[0] == '\0')
+		error (FATAL, "A parameter is needed after \"%s\" option", option);
+
+	patternLengthLimit = strtol(parameter, &end, 0);
```

Comment:
```
RETURN VALUE
       The strtol() function returns the result of the conversion, unless the value would underflow
       or  overflow.   If  an  underflow occurs, strtol() returns LONG_MIN.  If an overflow occurs,
       strtol() returns LONG_MAX.  In both cases, errno is set to ERANGE.  Precisely the same holds
       for strtoll() (with LLONG_MIN and LLONG_MAX instead of LONG_MIN and LONG_MAX).
```

Check the value of errno.